odkx-src/build-scripts.rst

```diff
@@ -148,7 +148,27 @@ To run the UI tests:
 
       ./gradlew clean testSnapshotBasicDebug connectedSnapshotBasicDebugAndroidTest
 
+Build Variants
+~~~~~~~~~~~~~~~~~~
+
```

Comment:
It would be good to start off with an introduction of why we use build variants.
"ODK-X has a modular framework design with inter-dependencies between various ODK-X tools. To manage accepting code contributions, bug fixes, and other enhancements that might effect other tools in a unforeseen way ODK-X uses a 3 staged release workflow that involves 3 branches.

- development: the branch where new development, upgrades, and features are contributed and tested
- demo: This branch is the last stable version of development. This provides a staging area used for testing before moving to an official release. This is where preview releases are staged.
- master: The current stable release.
"

Since the branches are a different stages of development the dependency libraries are at different stages of development. To make sure you are building against the correct dependencies you need to either checkout a local copy of the dependencies using the same branch name in the same directory OR you can adjust the build variant to match the dependencies causing Gradle and Android Studio to fetch the correct dependencies when compiling.